### PR TITLE
Initialize RequestTelemetry with correlation information

### PR DESCRIPTION
**Pull request by yantang-msft**
Initialize RequestTelemetry's correlation information based on the new Request-Id header.
If the running AspNetCore is 1.0, only the old event "Microsoft.AspNetCore.Hosting.BeginRequest" will be sent. We will create an activity which will be used by the dependency collector.
If the running AspNetCore is 2.0, both the old event and the new event "Microsoft.AspNetCore.Hosting.HttpRequestIn.Start" will be sent. In this case, we ignore the old event so we don't duplicate the RequestTelemetry. AspNetCore will also create an activity for us in this case.
We will also look for standard headers in case the calling service doesn't have the latest AI sdk.

### CODE REVIEW

File: src/Microsoft.ApplicationInsights.AspNetCore/Extensions/ApplicationInsightsExtensions.cs [hunk 1]
```diff
@@ -128,7 +128,6 @@ public static IServiceCollection AddApplicationInsightsTelemetry(this IServiceCo
             services.AddSingleton<ITelemetryInitializer, DomainNameRoleInstanceTelemetryInitializer>();
             services.AddSingleton<ITelemetryInitializer, ComponentVersionTelemetryInitializer>();
             services.AddSingleton<ITelemetryInitializer, ClientIpHeaderTelemetryInitializer>();
-            services.AddSingleton<ITelemetryInitializer, OperationIdTelemetryInitializer>();
             services.AddSingleton<ITelemetryInitializer, OperationNameTelemetryInitializer>();
             services.AddSingleton<ITelemetryInitializer, ApplicationInsights.AspNetCore.TelemetryInitializers.OperationCorrelationTelemetryInitializer>();
```
Comment (lmolkova) [inline src/Microsoft.ApplicationInsights.AspNetCore/Extensions/ApplicationInsightsExtensions.cs]: Please check OperationCorrelationTelemetryInitializer and remove operation Id, root id and parent Id initialization from there

File: src/Microsoft.ApplicationInsights.AspNetCore/project.json [hunk 2]
```diff
@@ -76,32 +76,33 @@
...
+                "Microsoft.ApplicationInsights.DependencyCollector": "2.4.0-beta1",
+                "Microsoft.ApplicationInsights.PerfCounterCollector": "2.4.0-beta1",
+                "Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel": "2.4.0-beta1"
             }
         },
         "netstandard1.6": {
             "dependencies": {
-                "NETStandard.Library": "1.6.0",
-                "System.Net.NameResolution": "4.0.0"
+                "NETStandard.Library": "1.6.1",
```
Comment (lmolkova) [inline src/Microsoft.ApplicationInsights.AspNetCore/project.json]: reference DependencyCollector here and make sure tests for DependenycCollector on netstandard1.6 are enabled

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs [hunk 3]
```diff
@@ -17,6 +21,8 @@ internal class HostingDiagnosticListener : IApplicationInsightDiagnosticListener
         private readonly TelemetryClient client;
         private readonly ICorrelationIdLookupHelper correlationIdLookupHelper;
         private readonly string sdkVersion;
+        private bool? isAspNetCore20;
```
Comment (lmolkova) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: could you make it static and initialize in place, so it don't have to be nullable, you can also remove property

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs [hunk 4]
```diff
@@ -34,31 +40,67 @@ public HostingDiagnosticListener(TelemetryClient client, ICorrelationIdLookupHel
...
+        {
+            get
+            {
+                if (isAspNetCore20.HasValue)
+                {
+                    return isAspNetCore20.Value;
+                }
+
+                var hostingVersion = typeof(WebHostBuilder).GetTypeInfo().Assembly.GetName().Version;
+                isAspNetCore20 = hostingVersion.Major >= 2;
```
Comment (lmolkova) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: 👍

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs [hunk 5]
```diff
@@ -99,6 +150,124 @@ public void OnDiagnosticsUnhandledException(HttpContext httpContext, Exception e
             this.OnException(httpContext, exception);
         }
 
+        private RequestTelemetry InitializeRequestTelemetryOnHttpRequestInStart(HttpContext httpContext)
+        {
+            var requestTelemetry = new RequestTelemetry();
+
+            if (Activity.Current == null)
```
Comment (lmolkova) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: store Activity.Current in local variable: it's AsyncLocal and accessing nultiple times it is not completely cheap

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs [hunk 6]
```diff
@@ -99,6 +150,124 @@ public void OnDiagnosticsUnhandledException(HttpContext httpContext, Exception e
...
+                }
+
+                StringValues xmsRequestId;
+                if (httpContext.Request.Headers.TryGetValue(RequestResponseHeaders.StandardParentIdHeader, out xmsRequestId))
+                {
+                    requestTelemetry.Context.Operation.ParentId = xmsRequestId;
+                }
+            }
+
+            requestTelemetry.Id = Activity.Current?.Id;
```
Comment (lmolkova) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: Activity.Current is never null at this point

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs [hunk 7]
```diff
@@ -99,6 +150,124 @@ public void OnDiagnosticsUnhandledException(HttpContext httpContext, Exception e
             this.OnException(httpContext, exception);
         }
 
+        private RequestTelemetry InitializeRequestTelemetryOnHttpRequestInStart(HttpContext httpContext)
+        {
+            var requestTelemetry = new RequestTelemetry();
+
+            if (Activity.Current == null)
+            {
+                AspNetCoreEventSource.Instance.LogHostingDiagnosticListenerOnHttpRequestInStartActivityNull();
```
Comment (lmolkova) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: this means something really strange happened and AspNEtCore did not create activity for us but sent the event. return, there is nothing else we can do

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs (hunk 6 above)
Comment (lmolkova) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: must not be null as you return before

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs [hunk 8]
```diff
@@ -99,6 +150,124 @@ public void OnDiagnosticsUnhandledException(HttpContext httpContext, Exception e
...
+                            activity.AddBaggage(baggageItem.Name, baggageItem.Value);
+                            if (!requestTelemetry.Context.Properties.ContainsKey(baggageItem.Name))
+                            {
+                                requestTelemetry.Context.Properties[baggageItem.Name] = baggageItem.Value;
+                            }
+                        }
+                    }
+                }
+            }
+            else if (httpContext.Request.Headers.TryGetValue(RequestResponseHeaders.StandardRootIdHeader, out standardRootId))
```
Comment (lmolkova) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: 1. I think it's almost exactly the same code as on line 179 in `InitializeRequestTelemetryOnHttpRequestInStart`. Could you have a separate function that parses legacy headers

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs [hunk 9]
```diff
@@ -34,31 +40,67 @@ public HostingDiagnosticListener(TelemetryClient client, ICorrelationIdLookupHel
...
+
+        /// <summary>
+        /// Diagnostic event handler method for 'Microsoft.AspNetCore.Hosting.HttpRequestIn.Start' event.
+        /// </summary>
+        [DiagnosticName("Microsoft.AspNetCore.Hosting.HttpRequestIn.Start")]
+        public void OnHttpRequestInStart(HttpContext httpContext)
+        {
+            if (this.client.IsEnabled())
+            {
+                var requestTelemetry = InitializeRequestTelemetryOnHttpRequestInStart(httpContext);
```
Comment (lmolkova) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: is there any point in having additional `InitializeRequestTelemetryOnHttpRequestInStart` method? 
It is used just once and makes it harder to read code. 

The same applies to `InitializeRequestTelemetryOnBeginRequest`

However if you prefer to keep it like this, I'm also fine with it.

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs [hunk 10]
```diff
@@ -99,6 +150,124 @@ public void OnDiagnosticsUnhandledException(HttpContext httpContext, Exception e
...
+        private void SetAppIdInResponseHeader(HttpContext httpContext, RequestTelemetry requestTelemetry)
+        {
+            IHeaderDictionary responseHeaders = httpContext.Response?.Headers;
+            if (responseHeaders != null &&
+                !string.IsNullOrEmpty(requestTelemetry.Context.InstrumentationKey) &&
+                (!responseHeaders.ContainsKey(RequestResponseHeaders.RequestContextHeader) || HttpHeadersUtilities.ContainsRequestContextKeyValue(responseHeaders, RequestResponseHeaders.RequestContextTargetKey)))
+            {
+                string correlationId = null;
+                if (this.correlationIdLookupHelper.TryGetXComponentCorrelationId(requestTelemetry.Context.InstrumentationKey, out correlationId))
+                {
```
Comment (lmolkova) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: OperationCorrelationTelemetryInitializer also queries correlatioIdLookupHelper for the same instrumentation key.
I wonder if code from OperationCorrelationTelemetryInitializer could go here and then we can just get rid of it?

File: src/Microsoft.ApplicationInsights.AspNetCore/project.json (hunk 2 above)
Comment (yantang-msft) [inline src/Microsoft.ApplicationInsights.AspNetCore/project.json]: Will do this in another PR, I have seen issues with unit tests if doing so.

File: src/Microsoft.ApplicationInsights.AspNetCore/Extensions/ApplicationInsightsExtensions.cs (hunk 1 above)
Comment (yantang-msft) [inline src/Microsoft.ApplicationInsights.AspNetCore/Extensions/ApplicationInsightsExtensions.cs]: Done

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs (hunk 3 above)
Comment (yantang-msft) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: Sure

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs (hunk 5 above)
Comment (yantang-msft) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: good to know

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs (hunk 7 above)
Comment (yantang-msft) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: I'll just remove this. It shouldn't happen

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs (hunk 6 above)
Comment (yantang-msft) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: Yep

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs (hunk 8 above)
Comment (yantang-msft) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: It's not the same. In new events code path, we create another activity. In old events code path, only one Activity is created for both old and new headers.

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs (hunk 9 above)
Comment (yantang-msft) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: I feel this part is a bit long. And I want to give it a function name so people know what it is doing.

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs (hunk 10 above)
Comment (yantang-msft) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: OperationCorrelationTelemetryInitializer set Request.Source, here it sets Request.Reponse.Header. But basically, they are doing similar things. I guess this part of code can be moved into OperationCorrelationTelemetryInitializer, and I'm willing to remove the additional "InitializeRequestTelemetryOnBeginRequest" in this case.
However, the logic here will override the value, not like other initializers which first checks if the value is null or empty. So I don't want to take the risk of moving it to the initializer.

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs [hunk 11]
```diff
@@ -78,7 +123,13 @@ public void OnEndRequest(HttpContext httpContext, long timestamp)
         public void OnHostingException(HttpContext httpContext, Exception exception)
         {
             this.OnException(httpContext, exception);
-            this.EndRequest(httpContext, Stopwatch.GetTimestamp());
+
+            // In AspNetCore 1.0, when an exception is unhandled, it will only send UnhandledException event but not EndRequest event, so we need to EndRequest at here.
+            // In AspNetCore 2.0, after sending UnhandledException, it will stop the created activity, which will send HttpRequestIn.Stop event, so we just EndRequest at there.
```
Comment (dnduffy) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: nitpick: "...send the HttpRequestIn.Stop event, so we will just end the request there."

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs [hunk 12]
```diff
@@ -77,6 +62,12 @@ public void OnHttpRequestInStart(HttpContext httpContext)
         {
             if (this.client.IsEnabled())
             {
+                if (Activity.Current == null)
+                {
+                    AspNetCoreEventSource.Instance.LogHostingDiagnosticListenerOnHttpRequestInStartActivityNull();
+                    return;
+                }
+
                 var requestTelemetry = InitializeRequestTelemetryOnHttpRequestInStart(httpContext);
```
Comment (lmolkova) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: I still believe  InitializeRequestTelemetryOnHttpRequestInStart and InitializeRequestTelemetryOnBeginRequest are almost the same:
Basically InitializeRequestTelemetryOnBeginRequest  is 
```
InitializeRequestTelemetryOnBeginRequest (context, timestamp)
{
  var a = new Activity(...);
  // get Request-Id and Correlation-Context headers
  a.Start();
  InitializeRequestTelemetryOnHttpRequestInStart (context, timestamp)
}
```

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs (hunk 11 above)
Comment (yantang-msft) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: Modified :)

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs (hunk 12 above)
Comment (yantang-msft) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: It's somehow consolidated. :)

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs [hunk 13]
```diff
@@ -14,9 +18,17 @@
     /// </summary>
     internal class HostingDiagnosticListener : IApplicationInsightDiagnosticListener
     {
+        /// <summary>
+        /// Determine whether the running AspNetCore Hosting version is 2.0 or higher. This will affect what DiagnosticSource events we receive.
+        /// To support AspNetCore 1.0 and 2.0, we listen to both old and new events.
+        /// If the running AspNetCore version is 2.0, both old and new events will be sent. In this case, we will ignore the old events.
+        /// </summary>
+        public static bool IsAspNetCore20 = typeof(WebHostBuilder).GetTypeInfo().Assembly.GetName().Version.Major >= 2;
```
Comment (Dmitry-Matveev) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: Should we stay more conservative and keep it "==2" in case of the breaking changes and new events in 3.0 that are not accounted for in our logic below?

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs [hunk 14]
```diff
@@ -34,31 +46,99 @@ public HostingDiagnosticListener(TelemetryClient client, ICorrelationIdLookupHel
...
-                IHeaderDictionary responseHeaders = httpContext.Response?.Headers;
-                if (responseHeaders != null &&
-                    !string.IsNullOrEmpty(requestTelemetry.Context.InstrumentationKey) &&
-                    (!responseHeaders.ContainsKey(RequestResponseHeaders.RequestContextHeader) || HttpHeadersUtilities.ContainsRequestContextKeyValue(responseHeaders, RequestResponseHeaders.RequestContextTargetKey)))
+                var activity = new Activity(ActivityCreatedByHostingDiagnosticListener);
+                var isActivityCreatedFromRequestIdHeader = false;
+
+                StringValues requestId;
+                StringValues standardRootId;
+                if (httpContext.Request.Headers.TryGetValue(RequestResponseHeaders.RequestIdHeader, out requestId))
```
Comment (Dmitry-Matveev) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: Is not-null `httpContext` a guarantee of `BeginRequest` event in diagnostic source?

Comment (yantang-msft) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: The instrumentation code will always send the httpContext, but I'm not sure whether it could be null at there. Since there is never a null check for httpContext, I think it's almost guaranteed.

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs (hunk 13 above)
Comment (yantang-msft) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: It's more likely AspNetCore will preserve the new events, so it's better to fall back to the new events.
Anyways, this should be removed when we upgrade the AspNetCore dependency to > 2.0 and only listening to new events. And if we don't restrict user using AspNetCore 3.0 with this library, it's equivalent to ">= 2" here.

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs (hunk 14 above)
Comment (dnduffy) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: The framework guarantees it.  No null check needed.

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs [hunk 15]
```diff
@@ -34,31 +46,99 @@ public HostingDiagnosticListener(TelemetryClient client, ICorrelationIdLookupHel
...
                 {
-                    string correlationId = null;
-                    if (this.correlationIdLookupHelper.TryGetXComponentCorrelationId(requestTelemetry.Context.InstrumentationKey, out correlationId))
+                    isActivityCreatedFromRequestIdHeader = true;
+                    activity.SetParentId(requestId);
+
+                    string[] baggage = httpContext.Request.Headers.GetCommaSeparatedValues(RequestResponseHeaders.CorrelationContextHeader);
```
Comment (dnduffy) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: nitpick: I would probably have cached httpContext.Request.Headers in a variable instead of dereferencing it each time.

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs [hunk 16]
```diff
@@ -78,7 +161,13 @@ public void OnEndRequest(HttpContext httpContext, long timestamp)
         public void OnHostingException(HttpContext httpContext, Exception exception)
         {
             this.OnException(httpContext, exception);
-            this.EndRequest(httpContext, Stopwatch.GetTimestamp());
+
+            // In AspNetCore 1.0, when an exception is unhandled, it will only send UnhandledException event but not EndRequest event, so we need to EndRequest at here.
```
Comment (dnduffy) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: nitpick: "...when an exception is unhandled it will only send the UnhandledException event, but not the EndRequest event, so we need to call EndRequest here."

File: src/Microsoft.ApplicationInsights.AspNetCore/project.json [hunk 17]
```diff
@@ -76,18 +76,18 @@
...
+        "type": "build"
+      }
+      //"Text.Analyzers": {
+      //    "version": "1.2.0-beta2",
+      //    "type": "build"
+      //},
+      //"XmlDocumentationComments.Analyzers": {
+      //    "version": "1.2.0-beta2",
+      //    "type": "build"
+      //},
```
Comment (dnduffy) [inline src/Microsoft.ApplicationInsights.AspNetCore/project.json]: nitpick: this formatting didn't need to change, especially not to become inconsistent.

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs (hunk 16 above)
Comment (yantang-msft) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: updated, thx

Comment (dnduffy) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: Please push your updates.

File: src/Microsoft.ApplicationInsights.AspNetCore/project.json (hunk 17 above)
Comment (yantang-msft) [inline src/Microsoft.ApplicationInsights.AspNetCore/project.json]: VS is so active in formatting things for you. :)

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs (hunk 14 above)
Comment (yantang-msft) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: The AspNetCore's instrumentation code doesn't have null check for Request and Headers, and also our code for Request and Method, so I suppose it's guaranteed.

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs (hunk 15 above)
Comment (yantang-msft) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: Updated.

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs (hunk 16 above)
Comment (yantang-msft) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: It's pushed. The update doesn't get reflected in this "outdated" conversation.

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs (hunk 13 above)
Comment (pharring) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: Why public? Shouldn't it be private? And readonly?

Comment (yantang-msft) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: Can change to internal readonly. Internal here for test purpose.
